### PR TITLE
Add clippy CI job

### PR DESCRIPTION
**Pull request by morenol**
#22

### CODE REVIEW

File: src/linalg/mod.rs [hunk 1]
```diff
@@ -76,6 +76,11 @@ pub trait BaseVector<T: RealNumber>: Clone + Debug {
     /// Get number of elevemnt in the vector
     fn len(&self) -> usize;
 
+    /// Returns true if the vector is empty.
+    fn is_empty(&self) -> bool {
```
Comment (morenol) [inline src/linalg/mod.rs:80]: I just implemented this to remove the warning, but not sure if this actually makes sense and has a use case for this library. We can just ignore that warning if this is not needed

File: .circleci/config.yml [hunk 2]
```diff
@@ -24,3 +30,14 @@ jobs:
...
+    docker:
+      - image: circleci/rust:latest
+    steps:
+      - checkout
+      - run:
+          name: Install cargo clippy
+          command: rustup component add clippy
+      - run:
+          name: Run cargo clippy
+          command: cargo clippy -- -Drust-2018-idioms -Dwarnings
```
Comment (VolodymyrOrlov) [inline .circleci/config.yml:43]: I found that Clippy ignores code behind optional features. The tool generates a bunch of additional errors when the flag `--all-features` is enabled. Hope this won't discourage you from finishing this PR :)
```suggestion
          command: cargo clippy --all-features -- -Drust-2018-idioms -Dwarnings
```

File: src/lib.rs [hunk 3]
```diff
@@ -64,6 +64,13 @@
 //! let y_hat = knn.predict(&x).unwrap();
 //! ```
 
+#![allow(
```
Comment (VolodymyrOrlov) [inline src/lib.rs:67]: Can you either move this block to the top, right under 
```
#![warn(missing_docs)]
#![warn(missing_doc_code_examples)]
```
or move this block close to yours?

**Review by VolodymyrOrlov — CHANGES_REQUESTED** [review]

**Comment by VolodymyrOrlov** [comment]
@morenol I like your idea of having a static code analysis tool and the idea of running every PR through this tool. 

Can I also ask you to update [developer's guide page](https://github.com/smartcorelib/smartcorelib.org/blob/development/user_guide/developer.md) with a list of steps every developer should do prior to submitting PR for a review? I am thinking:

1. Run all tests, make sure there are no errors
1. Fix code formatting: cargo fmt
1. Run clippy: cargo clippy --all-features -- -Drust-2018-idioms -Dwarnings

This might be a new section, e.g. Code Quality, or part of other section. 

This info will help developers understand better what steps he/she needs to follow to ensure proper code quality.

**Review by VolodymyrOrlov — APPROVED** [review]
LGTM

**Pull request merged**